File: Daily Challenge/C/Find Smaller Integer - Digits 14.1.2026.c
#include <stdio.h>
#include <limits.h>
#include <string.h>

int main() {
    int n;
    scanf("%d", &n);
    long arr[n];
    for(int i = 0; i < n; i++)
        scanf("%ld", &arr[i]);

    long k;
    scanf("%ld", &k);
    char kstr[20];
    sprintf(kstr, "%ld", k);

    long sum = 0;
    for(int i = 0; kstr[i]; i++) {
        char d = kstr[i];
        long min = LONG_MAX;
        int found = 0;

        for(int j = 0; j < n; j++) {
            char numstr[20];
            sprintf(numstr, "%ld", arr[j]);
            if(strchr(numstr, d)) {
                if(arr[j] < min) min = arr[j];
                found = 1;
            }
        }

        if(found) sum += min;
    }

    printf("%ld\n", sum);
    return 0;
}

File: Daily Challenge/Java/Find Smaller Integer - Digits 14.1.2026.java
import java.util.*;

public class Hello {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int[] arr = new int[n];
        for(int i = 0; i < n; i++) arr[i] = sc.nextInt();

        int k = sc.nextInt();
        String kstr = String.valueOf(k);

        int sum = 0;
        for(char d : kstr.toCharArray()) {
            int min = Integer.MAX_VALUE;
            boolean found = false;
            for(int num : arr) {
                if(String.valueOf(num).indexOf(d) != -1) {
                    min = Math.min(min, num);
                    found = true;
                }
            }
            if(found) sum += min;
        }

        System.out.println(sum);
    }
}